test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage.Tests.ps1

```diff
@@ -0,0 +1,400 @@
+pWeq/Rr5OpFb3rSotGTl84aIjk3hPd3uHujPQh8GO2EQ5k6p6ukk+a7gOUB+pH8fHihFl5L7pI0z
+yRp0FvbZo//hmACYMvINoy2EQxjYLh7QLeE4qEr8bkzJVgEURUvcUpyHFJT6PGzUMqGx/Wjh2jJc
+HfEDPMUDoTE/QRzLW7XrmQgJIRuHgPI/cqmOyvpEvuwdRhYyHEKktRO3tGjeflohDCyDW9bxOaJV
+ZP64KBordM28ZHCQbnSdU0I5us6qiFX2PiLlBzRMH2ftUNMYReioqZyR+Xv5wjaoydV3//BDMH8M
+1lh9GazUO8+OtzQEH0jiBi6ctlzFT8nNI2C+cOB9S3yMAjCEQa8wNzAfMAcGBSsOAwIaBBR96vF2
+OksttXT1kXf+aez9EzDlsgQU4ck78h0WTy01zHLwSKNWK4wFFQM=
+"
+
+    $dataEnciphermentCert = $dataEnciphermentCert -replace '\s',''
+    $certBytes = [Convert]::FromBase64String($dataEnciphermentCert)
+    $certLocation = Join-Path $TestDrive "ProtectedEventLogging.pfx"
+    [IO.File]::WriteAllBytes($certLocation, $certBytes)
+
+    return $certLocation
+}
+
+
+Describe "CmsMessage cmdlets and Get-PfxCertificate basic tests" -Tags "CI" {
+    
+    BeforeAll {
+        $certLocation = Create-TestCertificate
+        $certLocation | Should Not BeNullOrEmpty | Out-Null
+    }
+
+    It "Verify Get-PfxCertificate -FilePath" {
+        $cert = Get-PfxCertificate -FilePath $certLocation
+        $cert.Subject | Should Be "CN=MyDataEnciphermentCert"
+    }
+
+    It "Verify Get-PfxCertificate -LiteralPath" {
+        $cert = Get-PfxCertificate -LiteralPath $certLocation
+        $cert.Subject | Should Be "CN=MyDataEnciphermentCert"
+    }
+
+    It "Verify Get-PfxCertificate positional argument" {
+        $cert = Get-PfxCertificate $certLocation
+        $cert.Subject | Should Be "CN=MyDataEnciphermentCert"
+    }
+
+    It "Verify CMS message recipient resolution by path" -Skip:(!$IsWindows) {
+        $ers = $null
+        $recipient = [System.Management.Automation.CmsMessageRecipient] $certLocation
+        $recipient.Resolve($ExecutionContext.SessionState, "Encryption", [ref] $ers)
+
+        $recipient.Certificates[0].Subject | Should Match 'CN=MyDataEnciphermentCert'
+    }
+
+    It "Verify CMS message recipient resolution by cert" -Skip:(!$IsWindows) {
+        $ers = $null
+        $cert = Get-PfxCertificate $certLocation
+        $recipient = [System.Management.Automation.CmsMessageRecipient] $cert
+        $recipient.Resolve($ExecutionContext.SessionState, "Encryption", [ref] $ers)
+
+        $recipient.Certificates[0].Subject | Should Match 'CN=MyDataEnciphermentCert'
+    }
+
+    It "Verify a CMS message can be protected / unprotected" -Skip:(!$IsWindows) {
+        $protected = "Hello World","How are you?" | Protect-CmsMessage -To $certLocation
+        $protected.IndexOf("-----BEGIN CMS-----") | Should Be 0
+
+        $message = $protected | Get-CmsMessage
+        $message.Recipients[0].IssuerName | Should Be "CN=MyDataEnciphermentCert"
+
+        $decrypted = $message | Unprotect-CmsMessage -To $certLocation
+        $decrypted | Should Be "Hello World`r`nHow are you?"
+
+        $decrypted = $protected | Unprotect-CmsMessage -To $certLocation
+        $decrypted | Should Be "Hello World`r`nHow are you?"
+    }
+}
+
+
```

Comment:
Oh, I intentionally put 2 empty lines between `Describe` blocks, and 1 empty line between `It` blocks.